pages/notification-settings/NotificationSettings.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { FC, useState } from 'react';
+import { BreadCrumbs } from 'astro_2.0/components/BreadCrumbs';
+import { NavLink } from 'astro_2.0/components/NavLink';
+import { Toggle } from 'components/inputs/Toggle';
+import {
+  NotificationSettingsGroup,
+  NotificationSettingsItem,
+} from 'types/notification';
+import {
+  NOTIFICATION_SETTINGS_GROUPS_DATA,
+  NOTIFICATION_SETTINGS_DATA,
+} from 'mocks/notificationsData';
+import styles from './NotificationSettings.module.scss';
+
+interface NotificationSettingsProps {
+  settingGroups: NotificationSettingsGroup[];
+  settings: NotificationSettingsItem[];
+}
+
+const NotificationSettings: FC<NotificationSettingsProps> = ({
+  settingGroups = NOTIFICATION_SETTINGS_GROUPS_DATA,
+  settings = NOTIFICATION_SETTINGS_DATA,
+}) => {
+  const [settingsState, setSettingsState] = useState(settings);
+
+  // Switch one Group Notifications
+  const toggleSettingsSwitch = (id: string) => {
+    const settingsStateList = [...settingsState];
+
+    settingsStateList.map(item => {
```

Comment:
Can't we use array reduce or map function here to transform an array instead of mutating its content here?